src/generate_flow_metadata.py

```diff
@@ -0,0 +1,114 @@
+import inspect
+from collections import defaultdict
+from pkgutil import iter_modules
+from typing import Any, Dict, Generator
+
+from griffe.dataclasses import Docstring
+from griffe.docstrings.parsers import Parser, parse
+from prefect import Flow, flow, task
+from prefect.utilities.importtools import load_module
+
+from utils import skip_parsing, submit_updates
+
+skip_sections = {"parameters", "raises"}
+
+
+def parse_flow_docstring(flow: Flow) -> Dict[str, Any]:
+    """
+    Parses the docstring of a flow.
+    """
+    sections = parse(Docstring(flow.description), Parser.google)
+
+    # created serialized version of each docstring section
+    docstring_sections = defaultdict(list)
+
+    for section in sections:
+        section = section.as_dict()
+        section_value = section.get("value", [])
+        if section["kind"] in skip_sections:
+            continue
+        elif section["kind"] == "text":
+            docstring_sections["summary"] = section_value
+        elif section["kind"] == "returns":
+            docstring_sections[section["kind"]] = "\n".join(
+                return_obj.description for return_obj in section_value
+            )
+        elif (
+            section["kind"] == "admonition" and section_value["annotation"] == "example"
+        ):
+
+            docstring_sections["examples"].append(section_value["description"])
+
+    return docstring_sections
+
+
+def summarize_flow(flow: Flow, collection_name: str):
+    return {
+        "name": flow.name,
+        "parameters": dict(flow.parameters),
+        "install_command": f"pip install {collection_name}",
+        "description": {**parse_flow_docstring(flow)},
+    }
+
+
+def find_flows_in_module(
+    module_name: str,
+) -> Generator[Flow, None, None]:
+    """
+    Finds all flows in a module.
+    """
+    module = load_module(module_name)
+
+    # support loading flows from a module that is not a package
+    if not hasattr(module, "__path__"):
+        for _, obj in inspect.getmembers(module):
+            if isinstance(obj, Flow):
+                yield obj
+
+    else:
+        for _, name, ispkg in iter_modules(module.__path__):
+            if ispkg:
+
+                yield from find_flows_in_module(f"{module_name}.{name}")
+            else:
+                try:
+                    submodule = load_module(f"{module_name}.{name}")
+                    print(f"\t\tLoaded submodule {module_name}.{name}...")
+                except ModuleNotFoundError:
+                    continue
+
+                for _, obj in inspect.getmembers(submodule):
+                    if skip_parsing(name, obj, module_name):
+                        continue
+                    if isinstance(obj, Flow):
+                        yield obj
```

Comment:
This generator seems complex and has some redundant logic. I wonder if we can simplify this so it's easier to digest.

Reply:
sure, I'm open to suggestions - what do you see as redundant logic?

Reply:
The submodule branch of logic repeats some logic.

Reply:
can you elaborate?

Reply:
```

    # support loading flows from a module that is not a package
    if not hasattr(module, "__path__"):
        for _, obj in inspect.getmembers(module):
            if isinstance(obj, Flow):
                yield obj
```

```

                for _, obj in inspect.getmembers(submodule):
                    if skip_parsing(name, obj, module_name):
                        continue
                    if isinstance(obj, Flow):
                        yield obj
```
I believe is very similar and probably can be refactored into a separate function.